### PR TITLE
Improve color theme selection for navbars

### PR DESCRIPTION
**Pull request by luyangkenneth**
Based on the discussion in https://github.com/MarkBind/markbind/issues/386

### CODE REVIEW

File: src/Navbar.vue [hunk 1]
```diff
@@ -1,10 +1,11 @@
 <template>
   <nav ref="navbar" :class="['navbar', 'navbar-expand-md',
   {
-    'navbar-dark':(type === 'inverse'),
-    'navbar-light':(type === 'default'),
-    'bg-dark':(type === 'inverse'),
-    'bg-light':(type === 'default')
+    'navbar-dark':(type === 'primary' || type === 'dark' || type !== 'none'),
```
Comment (acjh) [inline src/Navbar.vue]: `type !== 'none'` includes `'light'` right?

Comment (luyangkenneth) [reply src/Navbar.vue]: This has been fixed!

File: src/Navbar.vue [hunk 2]
```diff
@@ -49,6 +43,28 @@ export default {
...
+          ? false
+          : true
+
+      return {
+        'navbar-dark': isNavbarDark,
+        'navbar-light': (this.type === 'light'),
+        'bg-primary': isBgPrimary,
+        'bg-dark': (this.type === 'dark'),
+        'bg-light': (this.type === 'light')
+      }
```
Comment (yamgent) [inline src/Navbar.vue]: 1. Prefer golden path rule whenever possible.
2. The equivalent of not using a valid theme keyword is the same as using the 'primary' theme.

```js
    themeOptions () {
      if (this.type === 'none') {
        return {};
      }
      
      const allowedThemes = ['primary', 'dark', 'light'];
      
      if (!allowedThemes.includes(this.type)) {
        this.type = 'primary';
      }
      
      return {
        'navbar-dark': (this.type === 'primary' || this.type === 'dark'),
        'navbar-light': (this.type === 'light'),
        'bg-primary': (this.type === 'primary'),
        'bg-dark': (this.type === 'dark'),
        'bg-light': (this.type === 'light')
      }
    } 
```

Comment (luyangkenneth) [reply src/Navbar.vue]: Ah, thanks for this. Should have occurred to me that I can do an early return with `type === none`.

Comment (acjh) [reply src/Navbar.vue]: A `switch` statement seems clearer.

```js
switch (this.type) {
  case 'dark':
    return 'navbar-dark bg-dark';
  ...
}
```

Comment (yamgent) [reply src/Navbar.vue]: > A `switch` statement seems clearer.
> 
> ```js
> switch (this.type) {
>   case 'dark':
>     return 'navbar-dark bg-dark';
>   ...
> }
> ```

Excellent point.

Comment (luyangkenneth) [reply src/Navbar.vue]: Nice, I've pushed the changes 👍

**Pull request merged**